PWGCF/Flow/Tasks/FlowRunbyRun.cxx

```diff
@@ -164,6 +225,7 @@ struct FlowRunbyRun {
     // detect run number
     auto bc = collision.bc_as<aod::BCsWithTimestamps>();
     int runNumber = bc.runNumber();
+    float l_Random = fRndm->Rndm();
     if (std::find(RunNumbers.begin(), RunNumbers.end(), runNumber) == RunNumbers.end()) {
```

Comment:
If the run number has not changed with respect to the previous collision there is not point for searching again for it
Please consider it for the next iteration